### PR TITLE
Add Data Plane RBAC functionality

### PR DESCRIPTION
**Pull request by sindhuba**
[Preview this branch](https://cosmos-explorer-preview.azurewebsites.net/pull/1878?feature.enableAadDataPlane=true)

### CODE REVIEW

File: src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx [hunk 1]
```diff
@@ -275,6 +285,22 @@ function createOpenSynapseLinkDialogButton(container: Explorer): CommandButtonCo
   };
 }
 
+function createLoginForEntraIDButton(container: Explorer): CommandButtonComponentProps {
```
Comment (jawelton74) [inline src/Explorer/Menus/CommandBar/CommandBarComponentButtonFactory.tsx:288]: We should only display the login button when we know that we are using RBAC and we don't have the token. i.e. the button should disappear after user has logged in.

File: src/Explorer/Panes/SettingsPane/SettingsPane.tsx [hunk 2]
```diff
@@ -383,6 +432,54 @@ export const SettingsPane: FunctionComponent<{ explorer: Explorer }> = ({
...
+          <>
+            <div className="settingsSection">
+              <div className="settingsSectionPart">
+                <fieldset>
+                  <legend id="enableDataPlaneRBACOptions" className="settingsSectionLabel legendLabel">
+                    Enable Entra ID RBAC
+                  </legend>
+                  <TooltipHost
+                    content={
+                      <>
```
Comment (jawelton74) [inline src/Explorer/Panes/SettingsPane/SettingsPane.tsx:445]: We should indicate that this only applies to NoSQL API accounts.

File: src/Explorer/Explorer.tsx [hunk 3]
```diff
@@ -251,8 +253,49 @@ export default class Explorer {
...
+        msalInstance.setActiveAccount(cachedAccount);
+        const aadToken = await acquireTokenWithMsal(msalInstance, {
+          forceRefresh: true,
+          scopes: [hrefEndpoint],
+          authority: `${configContext.AAD_ENDPOINT}${localStorage.getItem("cachedTenantId")}`,
+        });
+        updateUserContext({ aadToken: aadToken });
+      } catch (error) {
+        if (error instanceof msal.AuthError && error.errorCode === msal.BrowserAuthErrorMessage.popUpWindowError.code) {
+          useDialog
```
Comment (jawelton74) [inline src/Explorer/Explorer.tsx:282]: I think these errors can just go to the Notifications instead of opening a popup ?

Comment (sindhuba) [reply src/Explorer/Explorer.tsx:282]: sure, will fix it

File: src/Common/CosmosClient.ts [hunk 4]
```diff
@@ -157,7 +190,6 @@ export function client(): Cosmos.CosmosClient {
 
   const options: Cosmos.CosmosClientOptions = {
     endpoint: endpoint() || "https://cosmos.azure.com", // CosmosClient gets upset if we pass a bad URL. This should never actually get called
-    key: userContext.masterKey,
```
Comment (languy) [inline src/Common/CosmosClient.ts:160]: Does this affect the emulator?

File: src/Explorer/Panes/SettingsPane/SettingsPane.tsx [hunk 5]
```diff
@@ -44,6 +49,14 @@ export const SettingsPane: FunctionComponent<{ explorer: Explorer }> = ({
       ? Constants.Queries.UnlimitedPageOption
       : Constants.Queries.CustomPageOption,
   );
+
+  const [enableDataPlaneRBACOption, setEnableDataPlaneRBACOption] = useState<string>(
+    LocalStorageUtility.hasItem(StorageKey.DataPlaneRbacEnabled)
```
Comment (languy) [inline src/Explorer/Panes/SettingsPane/SettingsPane.tsx:54]: Does this caching work if I enable RBAC and disable later and re-enable again, etc?

**Review by languy — DISMISSED** [review]

File: src/Common/CosmosClient.ts (hunk 4 above)
Comment (sindhuba) [reply src/Common/CosmosClient.ts:160]: No, I don't think so - we're using EmulatorMasterKey which is hardcoded for the emulator Platform in the Token Provider.

File: src/Explorer/Panes/SettingsPane/SettingsPane.tsx (hunk 5 above)
Comment (sindhuba) [reply src/Explorer/Panes/SettingsPane/SettingsPane.tsx:71]: Yes, it did work as per my testing.

**Review by jawelton74 — DISMISSED** [review]

**Review by jawelton74 — DISMISSED** [review]

**Review by sunghyunkang1111 — APPROVED** [review]

**Review by jawelton74 — APPROVED** [review]

**Review by asierisayas — APPROVED** [review]

**Pull request merged**